students/piechowski.michal/lesson_03_functions/calculator.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+""" Simple python calculator. """
+
+def print_help():
+    """ Explains options used in the script. """
+    print("a - add")
+    print("s - subtract")
+    print("m - multiply")
+    print("d - divide")
+    print("p - power")
+    print("h,? - help")
+    print("q - QUIT")
+
+def print_result(result):
+    """ Prints given result. """
+    print("Result:")
+    print(result)
+
+def get_option():
+    """ Returns option choosen by the user. """
+    print("Enter option:")
+    option = input()
+    return option
+
+def get_two_operands():
+    """ Returns two operands used later for calculations. """
+    print("Input 1st operand:")
+    first_operand = float(input())
+    print("Input 2nd operand:")
+    second_operand = float(input())
+    return first_operand, second_operand
+
+def add():
+    """ Adds two numbers. """
+    print("ADDING")
+    first_operand, second_operand = get_two_operands()
+    print_result(first_operand + second_operand)
+
+def subtract():
+    """ Subtracts two numbers. """
+    print("SUBTRACT")
+    first_operand, second_operand = get_two_operands()
+    print_result(first_operand - second_operand)
+
+def multiply():
+    """ Multiplies two numbers. """
+    print("MULTIPLY")
+    first_operand, second_operand = get_two_operands()
+    print_result(first_operand * second_operand)
+
+def divide():
+    """ Divides two numbers. """
+    print("DIVIDE")
+    first_operand, second_operand = get_two_operands()
+    if second_operand == 0:
+        print("Can't divide by 0!")
+        return
+    print_result(first_operand / second_operand)
+
+def power():
+    """ Calculates first number to the power of second number. """
+    print("POWER")
+    first_operand, second_operand = get_two_operands()
+    print_result(first_operand ** second_operand)
+
+print("Welcome to nicely organized calculator:")
```

Comment:
całą tą część kodu, która już wywołuje kalkulator można by zamknąć w warunku:
`if __name__ == '__main__':`
Dzięki czemu podczas importu funkcji z tego modułu do innego modułu/skryptu ta część modułu nie zostanie wykonana.
On się wykona tylko wtedy, gdy ten moduł/skrypt zostanie uruchomiony jak skrypt (na najwyższym poziomie).
Jeżeli uruchomisz ten skrypt zmienna __name__ przyjmie wartość '__main__', jeżeli zaimportujesz jako moduł to __name__ ma wartość: 'nazwa_pliku'.

Reply:
W pozostałych zadaniach tez tego użyj